src/orion/client/experiment.py

```diff
@@ -617,19 +622,26 @@ def workon(self, fct, max_trials=infinity, **kwargs):
         """
         self._check_if_executable()
 
-        trials = 0
-        kwargs = flatten(kwargs)
-        while not self.is_done and trials < max_trials:
-            trial = self.suggest()
-            if trial is None:
-                log.warning("Algorithm could not sample new points")
-                return trials
-            kwargs.update(flatten(trial.params))
-            results = fct(**unflatten(kwargs))
-            self.observe(trial, results=results)
-            trials += 1
-
-        return trials
+        with Parallel(n_jobs=n_jobs, verbose=100) as parallel:
+            trials = parallel(
+                delayed(self._optimize)(fct, kwargs) for _ in range(max_trials)
```

Comment:
I think we could avoid lots of overhead by creating `n_jobs` instead and optimizing sequentially in each. 

```suggestion
                delayed(self._optimize)(fct, kwargs) for _ in range(n_jobs)
```

Reply:
Seems a good idea. But then maybe instead of `n_jobs`, `n_workers` is more accurate.

Reply:
Oh yes, I totally agree!